util/linuxfw/linuxfw.go

```diff
@@ -109,7 +109,8 @@ func getTailscaleSubnetRouteMark() []byte {
 // missing.  It does not check that IPv6 is currently functional or
 // that there's a global address, just that the system would support
 // IPv6 if it were on an IPv6 network.
-func CheckIPv6(logf logger.Logf) error {
+// CheckIPv6 can be overridden in tests.
+var CheckIPv6 = func(logf logger.Logf) error {
```

Comment:
I'd prefer this signature remain unchanged and instead add a new unexported package-level func that, if non-nil, is used inside the body, like:

```go
var checkIPv6ForTest func(logger.Logf) error // non-nil in some tests

func CheckIPv6(logf logger.Logf) error {
    if f := checkIPv6ForTest; f != nil {
        return f(logf)
    }
    ....
```

Then use tstest.Replace to set it.

Reply:
Oh that's really nice